sdk/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregations.java

```diff
@@ -0,0 +1,185 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.view;
+
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
+import io.opentelemetry.sdk.metrics.aggregator.DoubleSumAggregator;
+import io.opentelemetry.sdk.metrics.aggregator.LongSumAggregator;
+import io.opentelemetry.sdk.metrics.aggregator.NoopAggregator;
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.common.InstrumentValueType;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import javax.annotation.concurrent.Immutable;
+
+public class Aggregations {
+
+  /**
+   * Aggregation that calculates sum of recorded measurements.
+   *
+   * @since 0.1.0
+   */
+  public static Aggregation sum() {
+    return Sum.INSTANCE;
+  }
+
+  /**
+   * Aggregation that calculates count of recorded measurements (the number of recorded
+   * measurements).
+   *
+   * @since 0.1.0
+   */
+  public static Aggregation count() {
+    return Count.INSTANCE;
+  }
+
+  /**
+   * Aggregation that calculates distribution stats on recorded measurements. Distribution includes
+   * sum, count, histogram, and sum of squared deviations.
+   *
+   * <p>The boundaries for the buckets in the underlying histogram needs to be sorted.
+   *
+   * @param bucketBoundaries bucket boundaries to use for distribution.
+   * @return a new {@code Distribution}.
+   * @since 0.1.0
+   */
+  public static Aggregation distributionWithExplicitBounds(Double... bucketBoundaries) {
+    return new Distribution(bucketBoundaries);
+  }
+
+  /**
+   * Aggregation that calculates the last value of all recorded measurements.
+   *
+   * @since 0.1.0
+   */
+  public static Aggregation lastValue() {
+    return LastValue.INSTANCE;
+  }
+
+  @Immutable
+  private enum Sum implements Aggregation {
+    INSTANCE;
+
+    @Override
+    public AggregatorFactory getAggregatorFactory(InstrumentValueType instrumentValueType) {
+      return instrumentValueType == InstrumentValueType.LONG
+          ? LongSumAggregator.getFactory()
+          : DoubleSumAggregator.getFactory();
+    }
+
+    @Override
+    public Type getDescriptorType(
+        InstrumentType instrumentType, InstrumentValueType instrumentValueType) {
+      switch (instrumentType) {
+        case COUNTER_MONOTONIC:
+        case MEASURE_ABSOLUTE:
+        case OBSERVER_MONOTONIC:
+          return instrumentValueType == InstrumentValueType.LONG
+              ? Type.MONOTONIC_LONG
+              : Type.MONOTONIC_DOUBLE;
+        case COUNTER_NON_MONOTONIC:
+        case MEASURE_NON_ABSOLUTE:
+        case OBSERVER_NON_MONOTONIC:
+          return instrumentValueType == InstrumentValueType.LONG
+              ? Type.NON_MONOTONIC_LONG
+              : Type.NON_MONOTONIC_DOUBLE;
+      }
+      throw new IllegalArgumentException("This should not happen");
```

Comment:
change message to "Unsupported instrument/value types"

Reply:
done.